BoxContentSDK/BoxContentSDK/BOXURLSessionManager.m

```diff
@@ -536,7 +551,14 @@ - (BOXBackgroundSessionIdAndTask *)existingBackgroundSessionTaskGivenUserId:(NSS
 //      re-create the download task with resume data and return that
 - (NSURLSessionTask *)backgroundTaskWithRequest:(NSURLRequest *)request orUploadFromFile:(NSURL *)uploadFromFileURL taskDelegate:(id <BOXURLSessionTaskDelegate>)taskDelegate userId:(NSString *)userId associateId:(NSString *)associateId error:(NSError **)outError
 {
-    [self waitForBackgroundSessionSetUpToFinish];
+    if (self.finishSettingUpBackgroundSession == NO) {
+        //trying to get a background session task before setting up background session completes
+        //will fail this method
```

Comment:
this is unlikely. The app is expected to wait for completionBlock of the one time set up of background session to request for background tasks.